File: blaze_flutter_sdk/android/src/main/kotlin/com/blaze/flutterblazesdk/players/shared/BlazeReactCaptionsPositioningParser.kt
package com.blaze.flutterblazesdk.players.shared

import com.blaze.blazesdk.style.players.BlazeCaptionsXPosition
import com.blaze.blazesdk.style.players.BlazeCaptionsYPosition

/**
 * Parses a raw Map (deserialized from Flutter via Gson) into a native [BlazeCaptionsXPosition].
 *
 * The map comes in the form: { "type": "start" } or { "type": "custom", "offsetPercent": 50 }.
 */
fun Map<String, Any?>.toBlazeCaptionsXPosition(): BlazeCaptionsXPosition? {
    val type = this["type"] as? String ?: return null

    return when (type) {
        "start" -> BlazeCaptionsXPosition.Start
        "center" -> BlazeCaptionsXPosition.Center
        "custom" -> {
            val offsetPercent = (this["offsetPercent"] as? Number)?.toInt() ?: return null
            BlazeCaptionsXPosition.Custom(offsetPercent)
        }
        else -> null
    }
}

/**
 * Parses a raw Map (deserialized from Flutter via Gson) into a native [BlazeCaptionsYPosition].
 *
 * The map comes in the form: { "type": "top" } or { "type": "custom", "offsetPercent": 50 }.
 */
fun Map<String, Any?>.toBlazeCaptionsYPosition(): BlazeCaptionsYPosition? {
    val type = this["type"] as? String ?: return null

    return when (type) {
        "top" -> BlazeCaptionsYPosition.Top
        "center" -> BlazeCaptionsYPosition.Center
        "bottom" -> BlazeCaptionsYPosition.Bottom
        "custom" -> {
            val offsetPercent = (this["offsetPercent"] as? Number)?.toInt() ?: return null
            BlazeCaptionsYPosition.Custom(offsetPercent)
        }
        else -> null
    }
}

File: blaze_flutter_sdk/ios/blaze_flutter_sdk/Sources/Players/Shared/BlazeReactCaptionsPositioningParser.swift
//
//  BlazeReactCaptionsPositioningParser.swift
//  blaze_flutter_sdk
//
//

import Foundation
import BlazeSDK

extension BlazeCaptionsXPosition {
    static func from(dictionary: [String: AnyHashable]) -> Self? {
        guard let type = dictionary["type"] as? String else { return nil }

        switch type {
        case "start":
            return .start

        case "center":
            return .center

        case "custom":
            guard let offsetPercent = dictionary["offsetPercent"] as? Int else { return nil }
            return .custom(offsetPercent: offsetPercent)

        default:
            return nil
        }
    }
}

extension BlazeCaptionsYPosition {
    static func from(dictionary: [String: AnyHashable]) -> Self? {
        guard let type = dictionary["type"] as? String else { return nil }

        switch type {
        case "top":
            return .top

        case "center":
            return .center

        case "bottom":
            return .bottom

        case "custom":
            guard let offsetPercent = dictionary["offsetPercent"] as? Int else { return nil }
            return .custom(offsetPercent: offsetPercent)

        default:
            return nil
        }
    }
}
